crates/api/src/v1/package.rs

```diff
@@ -52,7 +52,7 @@ pub struct PublishRecordRequest<'a> {
     /// The complete set of content sources for the record.
     ///
     /// A registry may not support specifying content sources directly.
-    pub content_sources: HashMap<AnyHash, Vec<ContentSource>>,
+    pub content_sources: Option<HashMap<AnyHash, Vec<ContentSource>>>,
```

Comment:
If you want it to be optional for serialization purposes you can annotate it for serde:
```rust
#[serde(default, skip_serializing_if = "HashMap::is_empty")]
```

(`default` means "fill in with `Default` if missing)

Reply:
Ah, good call. I'll revise. Thanks.